systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java

```diff
@@ -2067,13 +2072,131 @@ void testLabelsAndAnnotationforPVC() {
+
+        LOGGER.info("Updating Kafka CR version field to " + newVersion.version());
+
+        // Get the Kafka resource from K8s
+        Kafka retrievedKafka = Crds.kafkaOperation(kubeClient(NAMESPACE).getClient())
+                .inNamespace(NAMESPACE)
+                .withName(CLUSTER_NAME)
+                .get();
+
+        // Change the Kafka version for the resource
+        retrievedKafka.getSpec().getKafka().setVersion(newVersion.version());
+
+        // Patch the existing resource with this new version
+        Crds.kafkaOperation(kubeClient().getClient()).inNamespace(NAMESPACE).withName(CLUSTER_NAME).patch(retrievedKafka);
+
+        LOGGER.info("Waiting for deployment of new Kafka version (" + newVersion.version() + ") to complete");
+
+        // Wait for the zk version change roll
+        zkPods = StUtils.waitTillSsHasRolled(KafkaResources.zookeeperStatefulSetName(CLUSTER_NAME), zkReplicas, zkPods);
+        LOGGER.debug("1st Zookeeper roll (image change) is complete");
+        // Wait for the zk rolling update
+        zkPods = StUtils.waitTillSsHasRolled(KafkaResources.zookeeperStatefulSetName(CLUSTER_NAME), zkReplicas, zkPods);
+        LOGGER.debug("2nd Zookeeper roll (update) is complete");
+
+        // Wait for the kafka broker version change roll
+        kafkaPods = StUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), kafkaReplicas, kafkaPods);
+        LOGGER.debug("Kafka roll (image change) is complete");
+
+        LOGGER.info("Deployment of Kafka (" + newVersion.version() + ") complete");
+
+        // Extract the zookeeper version number from the jars in the lib directory
+        zkResult = cmdKubeClient().execInPodContainer(KafkaResources.zookeeperPodName(CLUSTER_NAME, 0),
+                "zookeeper", "/bin/bash", "-c", zkVersionCommand).out().trim();
+        LOGGER.debug("Post-change Zookeeper version query returned: " + zkResult);
+
+        assertThat("Zookeeper container had version " + zkResult + " where " + newVersion.zookeeperVersion() +
+                " was expected", zkResult, is(newVersion.zookeeperVersion()));
+
+        // Extract the Kafka version number from the jars in the lib directory
+        kafkaResult = cmdKubeClient().execInPodContainer(KafkaResources.kafkaPodName(CLUSTER_NAME, 0),
+                "kafka", "/bin/bash", "-c", kafkaVersionCommand).out().trim();
+        LOGGER.debug("Post-change Kafka version query returned: " + kafkaResult);
+
+        assertThat("Kafka container had version " + kafkaResult + " where " + newVersion.version() +
+                " was expected", kafkaResult, is(newVersion.version()));
+
+        // TODO: If we added functionality tests, we could repeat them here to confirm the update resulted in a
+        //  functioning system and no data was lost?
+
+    }
+
+    @Test
+    void testKafkaClusterUpgrade() throws IOException {
```

Comment:
Could you move these new tests into `StrimziUpgradeST` ?

Reply:
These tests rely on the `createTestMethodResources()` and associated teardowns in the `KafkaST` `@BeforeEach` etc. I would have to manually add the setup and teardowns to each test if I move them. Is there a reasons you don't want them in the `KafkaST`? Would it make sense to create a `KafkaVersionChangeST`